protonvpn_linux_gui/presenters/dashboard_presenter.py

```diff
@@ -0,0 +1,565 @@
+import time
+import datetime
+import requests
+import subprocess
+import collections
+import concurrent.futures
+
+# Remote imports
+from protonvpn_cli.utils import (
+    get_config_value, 
+    is_connected, 
+    get_servers, 
+    get_server_value, 
+    get_transferred_data, 
+    pull_server_data,
+    get_country_name
+)
+from protonvpn_cli.country_codes import country_codes #noqa
+
+# PyGObject import
+import gi
+gi.require_version('Gtk', '3.0')
+from gi.repository import GObject as gobject, GdkPixbuf
+
+# Local imports
+from protonvpn_linux_gui.gui_logger import gui_logger
+from protonvpn_linux_gui.constants import GITHUB_URL_RELEASE, VERSION, LARGE_FLAGS_BASE_PATH, SMALL_FLAGS_BASE_PATH, FEATURES_BASE_PATH
+from protonvpn_linux_gui.utils import (
+    get_server_protocol_from_cli,
+    get_gui_config,
+    set_gui_config,
+    check_internet_conn,
+    custom_get_ip_info,
+)
+
+class DashboardPresenter:
+    def __init__(self, dashboard_service, queue):
+        self.dashboard_service = dashboard_service
+        self.queue = queue
+
+    def on_load(self, objects_dict):
+        """Calls load_on_start, which returns False if there is no internet connection, otherwise populates dashboard labels and server list
+        """
+        gui_logger.debug(">>> Running \"load_on_start\".")
+        return_val = False
+        display_message = "Could not load necessary resources, there might be connectivity issues."
+        time.sleep(2)
+        self.queue.put(dict(action="hide_spinner"))
+        
+        conn = custom_get_ip_info()
+
+        if conn and not conn is None:
+            self.queue.put(dict(action="update_dialog", label="Populating dashboard..."))
+            
+            display_secure_core = get_gui_config("connections", "display_secure_core")
```

Comment:
What are these operations and the following doing? I guess it would be nice to separate those into a dedicated function. 
https://www.todaysoftmag.com/article/1071/clean-code-functions, in particular _Do One Thing_ and _One level of abstraction per Function_